tests/src/test/java/alluxio/master/MasterFaultToleranceIntegrationTest.java

```diff
@@ -202,14 +228,10 @@ public void workerReRegister() throws Exception {
     AlluxioBlockStore store = AlluxioBlockStore.create();
     Assert.assertEquals(WORKER_CAPACITY_BYTES, store.getCapacityBytes());
 
-    List<Pair<Long, AlluxioURI>> emptyAnswer = new ArrayList<>();
     for (int kills = 0; kills < MASTERS - 1; kills++) {
       Assert.assertTrue(mMultiMasterLocalAlluxioCluster.stopLeader());
       mMultiMasterLocalAlluxioCluster.waitForNewMaster(CLUSTER_WAIT_TIMEOUT_MS);
-
-      // TODO(cc) Why this test fail without this line? [ALLUXIO-970]
-      faultTestDataCheck(emptyAnswer);
-
+      waitForWorkerRegistration(store, 1, Constants.SECOND_MS);
```

Comment:
Good point, updated. I originally made it short because the timing was only a few ms, but I think we will see fixed time overhead when running with other builds.